receiver/prometheusreceiver/internal/metadata.go

```diff
@@ -45,7 +45,7 @@ func (t *mService) Get(job, instance string) (MetadataCache, error) {
 
 	// from the same targetGroup, instance is not going to be duplicated
 	for _, target := range targetGroup {
-		if target.DiscoveredLabels().Get(model.AddressLabel) == instance {
```

Comment:
Comment for reviewers:

re: change from `DiscoveredLabels()` to `Labels()`: 
- `DiscoveredLabels()` contains "original" labels before relabelling
- `Labels()` contains modified labels after relabelling applied, see how two sets of labels are produced

So it seems that we receive updated labels (after relabelling) in https://github.com/open-telemetry/opentelemetry-collector/blob/c0dd66ef47bd94e7578a874583903059a2fe7925/receiver/prometheusreceiver/internal/transaction.go#L99 (the `instance` parameter is coming from there)?